### ISSUE TITLE
RF02, RF03 triggered on mysql for variables 

### ISSUE THREAD

**Issue by LucasCiocari** [sqlfluff/sqlfluff#5479]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Similar issue for T-SQL https://github.com/sqlfluff/sqlfluff/issues/1868, turned out to be a bug

When using variables inside a file, sqlfluff will trigger RF02 and RF03 for them thinking they need to be referenced by one of the tables.

Example: 

```sql
SET @someVar = 1;

SELECT
    Table1.Col1,
    Table2.Col2
FROM Table1
LEFT JOIN Table2 ON Table1.Join1 = Table2.Join1
WHERE Table1.FilterCol = @someVar;

_```___

### Expected Behaviour

No issues.

### Observed Behaviour

```sh
$ sqlfluff lint --dialect mysql teste.sql 
== [teste.sql] FAIL                                                                                                                                                                                  
L:   8 | P:  26 | RF02 | Unqualified reference '@someVar' found in select with
                       | more than one referenced table/view.
                       | [references.qualification]
All Finished 📜 🎉!
```

### How to reproduce

sqlfluff lint --dialect mysql teste.sql 

on 

```
SET @someVar = 1;

SELECT
    Table1.Col1,
    Table2.Col2
FROM Table1
LEFT JOIN Table2 ON Table1.Join1 = Table2.Join1
WHERE Table1.FilterCol = @someVar;

```

### Dialect

MySQL

### Version

2.3.5

### Configuration

[sqlfluff]
processes=-1
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = consistent
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = pascal
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = pascal
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = capitalise
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper
[sqfluff.indentation]
allow_implicit_indents=true
[sqlfluff:templater:jinja]

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mxcoder** [comment]
Same for BigQuery.
```
DECLARE mybar INT64 DEFAULT 1;

SELECT a.col1, b.col1, mybar
FROM a
INNER JOIN b
  ON a.col1 = b.col1
;
```
Will fail as well for RF02 due mybar not being "qualified".
